jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/TargetSystemIntegrationTest.java

```diff
@@ -0,0 +1,163 @@
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+
+public abstract class TargetSystemIntegrationTest {
+
+  private static final Logger logger = LoggerFactory.getLogger(TargetSystemIntegrationTest.class);
+  private static final String TARGET_SYSTEM_NETWORK_ALIAS = "targetsystem";
+  private static String otlpEndpoint;
+
+  /**
+   * Create target system container
+   *
+   * @param jmxPort JMX port target JVM should listen to
+   * @return target system container
+   */
+  protected abstract GenericContainer<?> createTargetContainer(int jmxPort);
+
+  private static Network network;
+  private static OtlpGrpcServer otlpServer;
+  private GenericContainer<?> target;
+  private JmxScraperContainer scraper;
+
+  private static final String OTLP_HOST = "host.testcontainers.internal";
+  private static final int JMX_PORT = 9999;
+
+  @BeforeAll
+  static void beforeAll() {
+    network = Network.newNetwork();
+    otlpServer = new OtlpGrpcServer();
+    otlpServer.start();
+    Testcontainers.exposeHostPorts(otlpServer.httpPort());
+    otlpEndpoint = "http://" + OTLP_HOST + ":" + otlpServer.httpPort();
+  }
+
+  @AfterAll
+  static void afterAll() {
+    network.close();
+    try {
+      otlpServer.stop().get();
+    } catch (InterruptedException | ExecutionException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @AfterEach
+  void afterEach() {
+    if (target != null && target.isRunning()) {
+      target.stop();
+    }
+    if (scraper != null && scraper.isRunning()) {
+      scraper.stop();
+    }
+    if (otlpServer != null) {
+      otlpServer.reset();
+    }
+  }
+
+  @Test
+  void endToEndTest() {
+
+    target =
+        createTargetContainer(JMX_PORT)
+            .withLogConsumer(new Slf4jLogConsumer(logger))
+            .withNetwork(network)
+            .withExposedPorts(JMX_PORT)
+            .withNetworkAliases(TARGET_SYSTEM_NETWORK_ALIAS);
+    target.start();
+
+    String targetHost = target.getHost();
+    Integer targetPort = target.getMappedPort(JMX_PORT);
+    logger.info(
+        "Target system started, JMX port: {} mapped to {}:{}", JMX_PORT, targetHost, targetPort);
+
+    // TODO : wait for metrics to be sent and add assertions on what is being captured
+    // for now we just test that we can connect to remote JMX using our client.
+    try (JMXConnector connector = JmxConnectorBuilder.createNew(targetHost, targetPort).build()) {
+      assertThat(connector.getMBeanServerConnection()).isNotNull();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    scraper =
+        new JmxScraperContainer(otlpEndpoint)
+            .withNetwork(network)
+            .withService(TARGET_SYSTEM_NETWORK_ALIAS, JMX_PORT);
+
+    scraper = customizeScraperContainer(scraper);
+    scraper.start();
+
+    // TODO: replace with real assertions
+    assertThat(otlpServer.getMetrics()).isEmpty();
```

Comment:
yeah so this is kinda weird. I appreciate the TODO, so we can circle back on it...but it's odd that we're doing all this setup to verify that nothing happened. :)

Reply:
Refactored a bit. Added abstract method to validate target system metrics. Method will be implemented in dedicated integration test.